base/floatfuncs.jl

```diff
@@ -304,7 +304,20 @@ true
 function isapprox(x::Number, y::Number;
                   atol::Real=0, rtol::Real=rtoldefault(x,y,atol),
                   nans::Bool=false, norm::Function=abs)
-    x == y || (isfinite(x) && isfinite(y) && norm(x-y) <= max(atol, rtol*max(norm(x), norm(y)))) || (nans && isnan(x) && isnan(y))
+    x′, y′ = promote(x, y) # to avoid integer overflow
```

Comment:
If we have the other method, why do we need to widen here?

Reply:
Because the other method is called only when both values are integer, but here `norm(x)` can overflow if `x` is an integer and `y` is a float. See tests added by the same commit.